scripts/passwordReset.groovy

```diff
@@ -0,0 +1,103 @@
+#!/usr/bin/env groovy
+/*
+    stream processor from stdin to stdout.
+    user records in ldif format is expected as input,
+    and it resets the password that it discovers to some random value.
+    
+    username & email address is recorded on the side into a separate file
+    specified as args[0] in tab-separated format.
+*/
+
+@Grab("commons-codec:commons-codec:1.10")
+import org.apache.commons.codec.binary.Base64;
+import java.security.*;
+
+MD = MessageDigest.getInstance("sha1")
+SR = new SecureRandom()
+
+main()
+System.exit(0)
+
+//================================================================
+
+
+// main loop
+def main() {
+	mail = null;
+	username = null;
+
+	// Our LDAP server has some additional special user accounts that shouldn't be reset
+	normalUser = false;
+
+	side = new PrintWriter(new FileWriter(args[0]));
+
+	System.in.eachLine { l ->
+		l = l.trim()
+		
+		// replace password
+		if (l.startsWith("userPassword:: ") && normalUser) {
+		  l = "userPassword:: "+base64(ssha(randomString(32),randomString(4)).bytes)
+		}
+		
+		System.out.println(l)
+		
+		
+		if (l=="") {
+		  if (mail!=null && username!=null) {
+		    side.println(mail+"\t"+username);
+		  }
+		  mail = null;
+		  username = null;
+		  normalUser = false;
+		  return;
```

Comment:
I can't remember my Groovy syntax well enough, but don't these `return` statements exit out of `main()`? Or do they just bounce to the next iteration of `eachLine`?